provider/prometheus.go

```diff
@@ -0,0 +1,133 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+
+	"github.com/evcc-io/evcc/provider/pipeline"
+	"github.com/evcc-io/evcc/util"
+	"github.com/prometheus/client_golang/api"
+)
+
+// Prometheus provider
+type Prometheus struct {
+	log         *util.Logger
+	client      api.Client
+	clientV1Api v1.API
+	query       string
+	scale       float64
```

Comment:
Where is the scale ever used? Is it needed when you can scale in the prom query?